src/paths/build_output_dir.py

```diff
@@ -8,5 +8,5 @@
 
 
 class BuildOutputDir(OutputDir):
-    def __init__(cls, filename, cwd=None, makedirs=True):
+    def __init__(cls, filename: str, cwd: str=None, makedirs: bool=True) -> None:
```

Comment:
Shouldn't the cwd be of type `Path`?

Reply:
Changed it to `Union[str, PathLike[str]]`

Reply:
I think this should stay as a string, in the unit tests the only type we use is string.  In the codebase its only reference is a string in `run_build.py`

Reply:
Note; the impact of this type is limited, whereas in https://github.com/opensearch-project/opensearch-build/pull/1564 its much broader in scope.

I would still prefer the return type changed to string

Reply:
Makes sense. @abhinavGupta16 are you changing it back to string?

Reply:
yes, I changed it all to `str` now